gobblin-config-management/src/main/java/gobblin/config/configstore/impl/BaseHdfsConfigStore.java

```diff
@@ -0,0 +1,214 @@
+      }
+
+      List<String> versions = new ArrayList<String>();
+      for (FileStatus f : fileStatus) {
+        // versions should be directory
+        if (!f.isDir()) {
+          continue;
+        }
+        versions.add(f.getPath().getName());
+      }
+      String res = this.vc.getCurrentVersion(versions);
+
+      if (res == null) {
+        throw new RuntimeException(String.format("location %s does not have any valid version: ", this.location));
+      }
+
+      return res;
+    } catch (IOException ioe) {
+      LOG.error("can not find current version: " + ioe.getMessage(), ioe);
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  @Override
+  public String getCurrentVersion() {
+    return this.currentVersion;
+  }
+  
+  @Override
+  public URI getStoreURI(){
+    return storeURI;
+  }
+  
+  @SuppressWarnings("deprecation")
+  protected Path getVersionRoot(String version){
+    Path versionRoot = new Path(this.location, version);
+    try {
+      if(this.fs.isDirectory(versionRoot)){
+        return versionRoot;
+      }
+      else {
+        String errorMesg = "Specified version " + versionRoot + " is not valid directory";
+        LOG.error(errorMesg);
+        throw new VersionDoesNotExistException(errorMesg);
+      }
+    } catch (IOException ioe) {
+      LOG.error("can not find the version " + version + " error:  " + ioe.getMessage(), ioe);
+      throw new VersionDoesNotExistException(ioe.getMessage());
+    } 
+  }
+
+  @Override
+  public Collection<URI> getChildren(URI uri, String version) {
+    if (uri == null)
+      return null;
+    
+    Path self = getPath(uri, version) ;
+    
+    try {
+      if (this.fs.isFile(self)) {
+        return Collections.emptyList();
+      }
+
+      FileStatus[] fileStatus = this.fs.listStatus(self);
+      if (fileStatus == null || fileStatus.length == 0) {
+        return Collections.emptyList();
+      }
+
+      List<URI> res = new ArrayList<URI>();
+      for (FileStatus f : fileStatus) {
+        // valid node should be a directory
+        if (!f.isDir()) {
+          continue;
+        }
+        try {
+          res.add(new URI(this.getRelativePath(f.getPath(), version)));
+        } catch (URISyntaxException e) {
+          LOG.error(String.format("Got error when create URI for %s, exception %s", f.getPath(), e.getMessage()), e);
+          throw new RuntimeException(e);
+        }
+      }
+
+      return res;
+    } catch (IOException ioe) {
+      LOG.error(String.format("Got error when find children for %s, exception %s", self, ioe.getMessage()));
+      throw new RuntimeException(ioe);
+    }
+  }
+  
+  @Override
+  public Config getOwnConfig(URI uri, String version) {
+    if (uri == null)
+      return ConfigFactory.empty();
+
+    Closer closer = Closer.create();
+    Path self = getPath(uri, version);
+    Path configFile = new Path(self, CONFIG_FILE_NAME);
+    try {
+      if (!this.fs.isFile(configFile)) {
+        return ConfigFactory.empty();
```

Comment:
This is the direct config for that node, so should return empty.

For getResolvedConfig, it will search ancestors